data_steward/analytics/cdr_ops/rt_cdr_qc/cdr_deid_qa_report6_fitdata.py

```diff
@@ -549,6 +552,53 @@
         ignore_index=True)
 df1.T
 
+# # Verify that all records with invalid level values are being dropped from sleep_level table
+#
+# DC-2606
+#
+# Queries the sandbox table for the corresponding cleaning rule (created in DC-2605) and outputs any records that are being dropped due to invalid level values. It also outputs any records in the sleep_level table that should have been dropped but were not.
+
+query = f'''
+
+WITH df1 AS (
+SELECT 
+*, 'dropped' AS dropped_status
+FROM 
+`{project_id}.{fitbit_sandbox_dataset}.{sleep_level_sandbox_table}`),
+
+df2 AS (
+SELECT 
+*,'not dropped' AS dropped_status
+FROM 
+`{project_id}.{fitbit_dataset}.sleep_level`
+WHERE 
+(
+    LOWER(level) NOT IN 
+         ('awake','light','asleep','deep','restless','wake','rem','unknown') OR level IS NULL))
+
+select *
+from df1
+UNION ALL
+select *
+from df2
+'''
+df2 = execute(client, query)
+if 'not dropped' not in set(df2['dropped_status']):
+    df = df.append(
+        {
+            'query': 'Query6 no invalid level records in sleep_level table',
+            'result': 'PASS'
+        },
+        ignore_index=True)
+else:
+    df = df.append(
+        {
+            'query': 'Query6 no invalid level records in sleep_level table',
+            'result': ''
+        },
+        ignore_index=True)
```

Comment:
This doesn't seem to include a failure or notification if we have a `dropped` status for something in that field.  I think it should have something if we are getting records we are not expecting and dropping them.  Maybe it's not a failure status but a "Follow up" status.

Reply:
This is the code that prints out a list that includes the status for each QA check in the notebook. The status for Query6 will be in that list

Reply:
`if not pass, will be highlighted in red
df = df.mask(df.isin(['Null', '']))
df.style.highlight_null(null_color='red').set_properties(
    **{'text-align': 'left'})`

Lines 613-615

This is the code that prints out the list that includes the status for each QA check in the notebook. The status for Query6 will be in that list.